src/hotspot/share/runtime/mutexLocker.hpp

```diff
@@ -265,6 +256,15 @@ class ConditionalMutexLocker: public MutexLockerImpl {
    }
 };
 
+// Same as MutexLocker but can be used during VM init.
+// Performs no action if given a null mutex or with detached threads.
+class NMTMutexLocker: public ConditionalMutexLocker {
```

Comment:
Good, but either call this something generic (e.g. "SafeMutexLocker") or move this to the nmt subdir.

Reply:
But this isn't generic as it locks the NMT_lock. Maybe move to NMT header file.

Reply:
Ok I've renamed the mutex and the locker to something more descriptive (`NmtVirtualMemory_lock` and `NmtVirtualMemoryLocker`). I've also moved the locker into nmtCommon.hpp